### PR TITLE
build(docs-infra): update docs examples to Angular v11.0.1

### PR DESCRIPTION
**Pull request by gkalpak**
This PR updates the docs examples to Angular v11.0.1. In addition to updating the dependencies versions, it also updates the project's structure and config to more closely match what a new v11 CLI app would look like. See, also, the [diff][1] between a basic v10.1.3 CLI app and a v11.0.2 one.

It also disables the Selenium Promise Manager (as seen [here][2]) and switches all e2e tests to `async/await` (to align with new CLI v11 apps).

<sub>(Supercedes #39787.)</sub>

[1]: https://github.com/cexbrayat/angular-cli-diff/compare/10.1.3..11.0.2
[2]: https://github.com/cexbrayat/angular-cli-diff/compare/10.1.3...11.0.2diff-dbd675d74087d57cd084d6dd6ae24ae2eeff2ff0122680e12916052f8a843a29

### CODE REVIEW

File: aio/content/examples/testing/src/app/demo/demo.testbed.spec.ts [hunk 1]
```diff
@@ -144,7 +144,7 @@ describe('demo (with TestBed):', () => {
     );
   });
 
-  describe('using async(inject) within beforeEach', () => {
+  describe('using waitForAsync(inject) within beforeEach', () => {
```
Comment (petebacondarwin) [inline aio/content/examples/testing/src/app/demo/demo.testbed.spec.ts:147]: Interesting that this bit of code does not appear in any guides!

**Review by petebacondarwin — APPROVED** [review]
Reviewed-for: docs-infra

**Review by petebacondarwin — APPROVED** [review]
Reviewed-for: global-docs-approvers

**Review by petebacondarwin — APPROVED** [review]
Reviewed-for: public_api

**Review by petebacondarwin — APPROVED** [review]
Reviewed-for: public-api

File: aio/content/examples/testing/src/app/demo/demo.testbed.spec.ts (hunk 1 above)
Comment (gkalpak) [reply aio/content/examples/testing/src/app/demo/demo.testbed.spec.ts:147]: It does (e.g. [here][1], [here][2] and [here][3]), but it was already updated there.

[1]: https://angular.io/guide/testing-components-basics
[2]: https://angular.io/guide/testing-components-scenarios
[3]: https://angular.io/guide/testing-utility-apis

Comment (petebacondarwin) [reply aio/content/examples/testing/src/app/demo/demo.testbed.spec.ts:147]: Not this specific line of code, it doesn't :-)

**Review by AndrewKushnir — APPROVED** [review]
Reviewed-for: public-api

File: aio/content/examples/testing/src/app/demo/demo.testbed.spec.ts (hunk 1 above)
Comment (gkalpak) [reply aio/content/examples/testing/src/app/demo/demo.testbed.spec.ts:147]: Oh, yeah, not this line of code. I meant `waitForAsync()` in general.
Overall, very few lines from e2e tests are included in the docs as code snippets.

**Review by IgorMinar — APPROVED** [review]
Thank you @gkalpak for the webdriver cleanup!! This is excellent. ❤️

**Review by IgorMinar — APPROVED** [review]
Reviewed-for: global-approvers

**Pull request closed without merging**